pkg/disk/partition_table_test.go

```diff
@@ -105,3 +105,552 @@ func TestPartitionTable_GenerateUUIDs_VFAT(t *testing.T) {
+			},
+			// no default fs required
+			expected: disk.PartitionTable{
+				Partitions: []disk.Partition{
+					{
+						Payload: &disk.Btrfs{
+							Subvolumes: []disk.BtrfsSubvolume{
+								{
+									Name:       "subvol/home",
+									Mountpoint: "/home",
+								},
+								{
+									Name:       "root",
+									Mountpoint: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"noop-lvm": {
+			pt: disk.PartitionTable{
+				Partitions: []disk.Partition{
+					{
+						Payload: &disk.LVMVolumeGroup{
+							Name: "testvg",
+							LogicalVolumes: []disk.LVMLogicalVolume{
+								{
+									Name: "varloglv",
+									Payload: &disk.Filesystem{
+										Label:      "var-log",
+										Type:       "xfs",
+										Mountpoint: "/var/log",
+									},
+								},
+								{
+									Name: "datalv",
+									Payload: &disk.Filesystem{
+										Label:        "data",
+										Type:         "ext4",
+										Mountpoint:   "/data",
+										FSTabOptions: "defaults",
+									},
+								},
+								{
+									Name: "rootlv",
+									Payload: &disk.Filesystem{
+										Label:        "root",
+										Type:         "ext4",
+										Mountpoint:   "/",
+										FSTabOptions: "defaults",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: disk.PartitionTable{
+				Partitions: []disk.Partition{
+					{
+						Payload: &disk.LVMVolumeGroup{
+							Name: "testvg",
+							LogicalVolumes: []disk.LVMLogicalVolume{
+								{
+									Name: "varloglv",
+									Payload: &disk.Filesystem{
+										Label:      "var-log",
+										Type:       "xfs",
+										Mountpoint: "/var/log",
+									},
+								},
+								{
+									Name: "datalv",
+									Payload: &disk.Filesystem{
+										Label:        "data",
+										Type:         "ext4",
+										Mountpoint:   "/data",
+										FSTabOptions: "defaults",
+									},
+								},
+								{
+									Name: "rootlv",
+									Payload: &disk.Filesystem{
+										Label:        "root",
+										Type:         "ext4",
+										Mountpoint:   "/",
+										FSTabOptions: "defaults",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"noop-btrfs": {
+			pt: disk.PartitionTable{
+				Partitions: []disk.Partition{
+					{
+						Payload: &disk.Btrfs{
+							Subvolumes: []disk.BtrfsSubvolume{
+								{
+									Name:       "subvol/home",
+									Mountpoint: "/home",
+								},
+								{
+									Name:       "root",
+									Mountpoint: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: disk.PartitionTable{
+				Partitions: []disk.Partition{
+					{
+						Payload: &disk.Btrfs{
+							Subvolumes: []disk.BtrfsSubvolume{
+								{
+									Name:       "subvol/home",
+									Mountpoint: "/home",
+								},
+								{
+									Name:       "root",
+									Mountpoint: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"err-empty": {
```

Comment:
(Super nitpick) *maybe* spliting into happy and unhappy tests might make things feel a little bit more organized but that might be just me (and slightly less struct/ content for each). I generally like mixing when the test cases are not overly long but when they are multiple pages like here it might be slightly nicer to split (but again could be preferences)

Reply:
I did at first but then wanted to deduplicate the common bits between the two.  Which isn't a lot, admittedly.